layer.go

```diff
@@ -173,9 +188,24 @@ func (s *Layer) Run(phase string, w http.ResponseWriter, r *http.Request, h http
 		h = queue[i](h)
 	}
 
-	// Memoize the phase trigger function
-	s.memo[phase] = h
-
-	// Trigger the first handler
+	// Trigger the first middleware handler
 	h.ServeHTTP(w, r)
 }
+
+// runRecoverError runs the current layer error phase middleware chain
+// triggering the parent layer if necessary.
+func (s *Layer) runRecoverError(rerr interface{}, w http.ResponseWriter, r *http.Request) {
+	next := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// If no parent, run default the final handler
+		if s.parent == nil {
+			FinalErrorHandler.ServeHTTP(w, r)
+			return
+		}
+		// If parent layer, trigger it
+		s.parent.Run("error", w, r, FinalErrorHandler)
+	})
+
+	// Expose error via context. This may change in a future.
+	context.Set(r, "error", rerr)
```

Comment:
If we use context, I would probably use the `vinxi.error` or anything else prefixed with `vinxi`. This way we're pretty sure nobody else would ever want to override it (Based on [how sinatra handles error](http://www.sinatrarb.com/intro.html#Error)).